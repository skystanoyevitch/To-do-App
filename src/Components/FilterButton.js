import React from "react";

export default function FilterButton(props) {
	return (
		<>
			<button
				type="button"
				aria-pressed={props.isPressed}
				onClick={() => {
					props.setFilter(props.name);
				}}
			>
				Show {props.name} tasks
			</button>
		</>
	);
}
