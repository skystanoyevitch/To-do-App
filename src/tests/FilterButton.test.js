import React from "react";
import renderer from "react-test-renderer";
import FilterButton from "../Components/FilterButton";

test("Button changes name when clicked", () => {
	const component = renderer.create(
		<FilterButton
			onClick={() => {
				props.setFilter(props.name);
			}}
		/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
