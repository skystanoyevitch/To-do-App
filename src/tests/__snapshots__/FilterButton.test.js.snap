// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes name when clicked 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Show 
   tasks
</button>
`;
